### ISSUE TITLE
Segfault when desired size is less than original size

### ISSUE THREAD

**Issue by slavistan** [seebye/ueberzug#57]
**tl;dr** When the *height* and *width* parameters are chosen such that the image would be shrunk with respect to its original size a segmentation fault occurs.

Here's a self-contained script which crashes when run. An image is downloaded via *curl* for your convenience.

```bash
#!/usr/bin/bash

url="https://images.pexels.com/photos/356378/pexels-photo-356378.jpeg"
curl "$url" --output "img.jpg" --silent

source "`ueberzug library`"

ImageLayer 0< <(
  ImageLayer::add [identifier]="foo" [path]="img.jpg" [x]=0 [y]=0 \
		[width]=500 [height]=500 # 1000 does not crash
  sleep 2
  ImageLayer::remove [identifier]="foo"
)
```
Running the script `./crash.sh` yields the following output:

```bash
/usr/local/lib64/python3.7/site-packages/ueberzug/lib/lib.sh: line 41:  5970 Segmentation fault      (core dumped) ueberzug layer -p bash "$@"
```
From experimenting a little bit it seems to me, that the condition causing the crash is that the desired image dimensions be smaller than the image's actual size (take that with a grain of salt - I might be mistaken). Images are displayed just fine when the *width* and *height* parameters are sufficiently large or when they're omitted altogether. 

Any ideas what could be going on? I'd be happy to help.

**Edit:** I got it "working" by replacing `pillow-simd` with `pillow` and using `ueberzug-nosimd` as suggested in #56. However, I do not understand wherein the problem lies. As this problem has been encountered by quite a few users already I kindly suggest that a comment shall be added to the project's readme.

**Comment by slavistan** [comment]
I just noticed that I'm not the only one. Gonna check the closed issues and report back.

**Comment by INIROBO** [comment]
Thanks for your quick fix slavistan. I got it to work with ueberzug-nosimd on Gentoo. And you are right, it crashes on 1000x1000 (but I didn't test it more to see if lowering it would still cause it to crash). Anyways my state remains equal for the common ueberzug and pillow-simd packages. Your _crash.sh_ file only crashes with SegFault on my Gentoo machine.

This is the post I was writing about the problem I was having, but I guess it is kind of solved now, but I hope it can be fixed on a future update.

And I emphasize, **on MX Linux works** and **on Gentoo it doesn't** with the same packages and scripts. I might be mistaken but I have uninstalled 2 times both Pillow(-simd) and uberzug(-nosimd) and reinstall them again to test whether they crash, but I didn't have any _luck_ (I guess).

> Hello,
> 
> I've seen your post about 18.1.0 ueberzug release. You also comment about some scripts you quickly made to be able to use it with vifm.
> 
> Ok, so I used your configurations in 2 machines. One uses MX Linux and the other Gentoo.
> 
> Vifm with your scripts on MX Linux works just great. I just run them and they worked. Unfortunatly, it wasn't the same in the case of Gentoo.
> 
> I think that it might be a pipe problem but I'm not sure because it seems that the declare statement on the bash script is piping the output correctly:
> ![Screenshot-2019-07-02-3](https://user-images.githubusercontent.com/32362244/60521905-5d1c6b00-9ce8-11e9-9554-4ece340f435f.png)
> 
> But ueberzug works as expected because if I use the bash example you provided on your github page (the one you name as "Bash library"), it works. It displays the image correctly on the terminal.
> 
> If I remove the "--silent" flag from ueberzug command on the bash script to run vifm, this is what I get:
> ![Screenshot-2019-07-02](https://user-images.githubusercontent.com/32362244/60521904-5c83d480-9ce8-11e9-84dc-661328fcf9c2.png)
> 
> After quitting Vifm, this message is print out (I think it gets printed before closing vifm):
> ![Screenshot-2019-07-02-4](https://user-images.githubusercontent.com/32362244/60521907-5d1c6b00-9ce8-11e9-8ed4-914d06f8a055.png)

MX Linux: xterm & xfce4-terminal
Gentoo Linux: st & xterm (I didn't try crash.sh on xterm)

**Comment by slavistan** [comment]
> And I emphasize, on MX Linux works and on Gentoo it doesn't with the same packages and scripts. I might be mistaken but I have uninstalled 2 times both Pillow(-simd) and uberzug(-nosimd) and reinstall them again to test whether they crash, but I didn't have any luck (I guess).

That is indeed very curious. It smells like an issue with pillow-simd on some machines/distros as at least another user in #56 has trouble running it by itself. We'll have to check again at a later point in time. I personally don't use pillow or pillow-simd for anything else aside ueberzug so I'm perfectly happy with the quickfix.
